oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2ClientAuthenticationFilter.java

```diff
@@ -166,6 +171,7 @@ private void onAuthenticationSuccess(HttpServletRequest request, HttpServletResp
 		SecurityContext securityContext = SecurityContextHolder.createEmptyContext();
 		securityContext.setAuthentication(authentication);
 		SecurityContextHolder.setContext(securityContext);
+		logger.debug(LogMessage.format("Set SecurityContextHolder to %s", securityContext));
```

Comment:
Let's avoid passing objects that will result in `toString()` since this may log sensitive information. For example, a custom `SecurityContext` may return sensitive information in it's `toString()`. Instead, let's log the `Authentication.getClass().getSimpleName()`